bin/bnf_to_cnf/bnf_to_cnf/driver.py

```diff
@@ -39,7 +40,9 @@ def __init__(self):
         self.translator = Translator()
         self.tree: Optional[Node] = None
 
-    def read(self, filename: str) -> "Driver":
+    def read(
+        self, filename: Union[int, str, bytes, PathLike[str], PathLike[bytes]]
```

Comment:
Is each of these types for `filename` actually going to be used when calling `Driver.read()`? Further down in this file I only see a `Path` object being used. Maybe just require that type for simplicity if it's not necessary to support the other types?

Reply:
> Maybe just require that type for simplicity if it's not necessary to support the other types?

Yeah, that's possible. I like to specify the broadest range of types so that one use the broader signature later without checking the entire code whether it supports set types.

Reply:
On the other hands, supporting a broader signature would strictly speaking require writing unit tests for each and every type supported. But fair enough, I can see the benefits of your approach as well.

Reply:
> On the other hands, supporting a broader signature would strictly speaking require writing unit tests for each and every type supported.

Right. Then I will reduce the type hint and add a note to the docstring or as a code comment.